thunder/distributed/__init__.py

```diff
@@ -349,6 +350,162 @@ def f(tensor: TensorProxy) -> str:
     return f
 
 
+def fsdp_transform_module(
+    thunder_model: "thunder.ThunderModule",
+    *,
+    device: torch.device | None = None,
+    broadcast_from: int | None = None,
+    sharding_strategy: FSDPType = FSDPType.ZERO2,
+    bucketing_strategy: FSDPBucketingStrategy = FSDPBucketingStrategy.NONE,
+) -> "thunder.ThunderModule":
+    import thunder
+
+    cd = thunder.compile_data(thunder_model)
+    cd.use_fsdp = True
```

Comment:
wouldn't it be high time we promoted `use_fsdp` and `use_ddp` to an public member of
 https://github.com/Lightning-AI/lightning-thunder/blob/ef3bcd07e55579e2b3d57da75aabb38fe1ae6862/thunder/common.py#L139-L158?

Reply:
added a todo